run_idempotent_azure.sh

```diff
@@ -36,16 +36,25 @@ modified_slug_module="${modifiedslug_with_sha}=${modified_module}"
 MVNOPTIONS="-Ddependency-check.skip=true -Dmaven.repo.local=$AZ_BATCH_TASK_WORKING_DIR/dependencies_${modifiedslug_with_sha}=${modified_module} -Dgpg.skip=true -DfailIfNoTests=false -Dskip.installnodenpm -Dskip.npm -Dskip.yarn -Dlicense.skip -Dcheckstyle.skip -Drat.skip -Denforcer.skip -Danimal.sniffer.skip -Dmaven.javadoc.skip -Dfindbugs.skip -Dwarbucks.skip -Dmodernizer.skip -Dimpsort.skip -Dmdep.analyze.skip -Dpgpverify.skip -Dxml.skip -Dcobertura.skip=true -Dfindbugs.skip=true"
 
 # echo "================Cloning the project"
-bash $dir/clone-project.sh "$slug" "$modified_slug_module" "$input_container"
+bash $dir/clone-project.sh "$slug" "${modifiedslug_with_sha}=${modified_module}" "$input_container"
 ret=${PIPESTATUS[0]}
 if [[ $ret != 0 ]]; then
     if [[ $ret == 2 ]]; then
         echo "$line,${modifiedslug_with_sha}=${modified_module},cannot_clone" >> $AZ_BATCH_TASK_WORKING_DIR/$input_container/results/"${modifiedslug_with_sha}=${modified_module}-results".csv
+        echo "Couldn't download the project. Actual: $ret"
+        exit 1
     elif [[ $ret == 1 ]]; then
-        echo "$line,${modifiedslug_with_sha}=${modified_module},cannot_checkout" >> $AZ_BATCH_TASK_WORKING_DIR/$input_container/results/"${modifiedslug_with_sha}=${modified_module}-results".csv
-    fi  
-    echo "Compilation failed. Actual: $ret"
-    exit 1
+        cd ~/
+        wget "https://github.com/$slug/archive/$sha".zip
+        ret=${PIPESTATUS[0]}
+        if [[ $ret != 0 ]]; then
+            echo "$line,${modifiedslug_with_sha}=${modified_module},failed_wget" >> $AZ_BATCH_TASK_WORKING_DIR/$input_container/results/"${modifiedslug_with_sha}=${modified_module}-results".csv
+            echo "Compilation failed. Actual: $ret"
+            exit 1
+        else
+            echo "$line,${modifiedslug_with_sha}=${modified_module},passed_wget" >> $AZ_BATCH_TASK_WORKING_DIR/$input_container/results/"${modifiedslug_with_sha}=${modified_module}-results".csv
```

Comment:
This is not enough for passed_wget. We did not install the project yet! You can set a variable if wget succeeded. If wget failed then you should be using ```cannot_checkout_or_wget```. AFTER install-project.sh is run you can then check whether wget succeeded or not. If wget succeeded then if install project succeeded the result is passed_wget, otherwise install project failed and the result is failed_wget.

Reply:
For this else clause specifically, you can just echo a message such as ```echo "git checkout failed but wget successfully downloaded the project and sha, proceeding to the rest of this script"```